ByteBuffer: Make less often senseless copies

BenWiederhake opened the pull request:
A while ago, I noticed that `ByteBuffer` has an implicit copy-constructor. So if you have a 10 Megabyte ByteBuffer, it's really easy to accidentally create lots and lots of copies of it, without noticing it.

I thought I'd "fix" this problem by just deleting the copy-constructor, replacing it by an explicit `copy() const` method, and be done with it. But alas, there are many problems:
- This bug is pretty pervasive, and this PR only fixes *very few* of the bugs instances.
- Some places even rely on this implicit copying, most prominently at least two places do `HashMap<T, ByteBuffer> m_foo; m_foo.get(…)`.

So maybe it there should be a `ByteBufferView` type, of maybe we should specialize `Traits<ByteBuffer>::PeekType` to be something different (`Bytes` probably?), or `get` shouldn't copy in the first place, or something like that. This discussion is out of scope for this PR, but I'd like to know how we wanna solve this. Let's discuss it in #ak.

This PR fixes the issue in *some* places. But not all. But at least it's progress.

IdanHo commented on Userland/Libraries/LibCore/MimeData.h:
```diff
@@ -20,7 +20,7 @@ class MimeData : public Object {
 
     ByteBuffer data(const String& mime_type) const { return m_data.get(mime_type).value_or({}); }
-    void set_data(const String& mime_type, const ByteBuffer& data) { m_data.set(mime_type, data); }
+    void set_data(const String& mime_type, ByteBuffer&& data) { m_data.set(mime_type, data); }
```
`move()`?

IdanHo reviewed (commented):
Looks good, just one comment

awesomekling commented:
There's a test failure in LibTLS/TestTLSHandshake on one of the builders. Any chance that's a regression from this, or just a flaky test? :)

BenWiederhake commented:
Changes since last push:
- Rebased
- Resolved a somewhat trivial conflict in LibSQL
- Do a `move` in MimeData do actually avoid copying (thanks IdanHo!), this kinda demonstrates how easy it is to accidentally copy a bytebuffer

I looked intensely at the changes to crypto-stuff, and I'm convinced that the CI failure was only a fluke.
- The changes to TestAES.cpp and test-crypto obviously are unrelated (or "really really should be")
- In TLSv12.cpp, we move out of a newly-constructed Optional, and nobody else uses this Optional. This shouldn't be problematic.
